tbot_contrib/locking.py

```diff
@@ -0,0 +1,166 @@
+import contextlib
+import typing
+import tbot
+import time
+from tbot.machine import machine, linux
+
+
+class MachineLock(machine.PreConnectInitializer):
+    """
+    Machine locking implementation based on Python and bash.
+    """
+    @property
+    def lock_name(self) -> str:
+        return self.__class__.__name__
+
+    @property
+    def lock_expiry(self) -> int:
+        return -1
```

Comment:
In python, we generally use `None` to mark the absence of a value instead of e.g. `-1`.  You can then check for it with

```python
if variable is not None:
    ...

# or

if variable is None:
    ...
```